### ISSUE TITLE
Toolchain build failing for 2014.12 tag on Linux

### ISSUE THREAD

**Issue by calvinatintel** [foss-for-synopsys-dwc-arc-processors/toolchain#60]
Hello,

Was the tag moved recently? Our daily automated builds have failed since May 23rd.
I would appreciate your help.

Thank you,
Calvin

```
$ mkdir arc_toolchain
sys_maker@mkslavel64:~$ cd arc_toolchain/
sys_maker@mkslavel64:~/arc_toolchain$ git clone https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain.git
Cloning into 'toolchain'...
remote: Counting objects: 2408, done.
remote: Total 2408 (delta 0), reused 0 (delta 0), pack-reused 2408
Receiving objects: 100% (2408/2408), 1.56 MiB | 1.01 MiB/s, done.
Resolving deltas: 100% (1658/1658), done.
Checking connectivity... done.
sys_maker@mkslavel64:~/arc_toolchain$ cd toolchain/
sys_maker@mkslavel64:~/arc_toolchain/toolchain$ git checkout arc-2014.12
Note: checking out 'arc-2014.12'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e9686ba... Create arc-versions.sh for tag arc-2014.12
sys_maker@mkslavel64:~/arc_toolchain/toolchain$ ./arc-clone-all.sh 
Cloned directories will be created in /home/sys_maker/arc_toolchain
Checking we can write in /home/sys_maker/arc_toolchain
Cloning cgen...
- successfully cloned ARC cgen repository
Cloning binutils...
- successfully cloned ARC binutils repository
Cloning gcc...
- successfully cloned ARC gcc repository
Cloning gdb...
- successfully cloned ARC gdb repository
Cloning newlib...
- successfully cloned ARC newlib repository
Cloning uClibc...
- successfully cloned ARC uClibc repository
Cloning linux...
- successfully cloned ARC linux repository
All repositories cloned
- full logs in /home/sys_maker/arc_toolchain/logs-4.8/clone-all-2015-05-29-2231.log
sys_maker@mkslavel64:~/arc_toolchain/toolchain$ ./build-all.sh --cpu arcem --no-uclibc --sim --no-pdf --rel-rpaths --config-extra "--enable-sim-endian=no --with-python=no LDFLAGS=-static"
Checking out GIT trees ...
ERROR: Failed to checkout GIT versions of tools
- see /home/sys_maker/arc_toolchain/logs-4.8/all-build-2015-05-29-2322.log
sys_maker@mkslavel64:~/arc_toolchain/toolchain$ 
```

At the end of `all-build-2015-05-29-2322.log` is this:

```
Checking out GIT trees
======================
Checking out branch/tag arc-2014.12 of cgen
  fetching branches
  fetching tags
  checking out arc-2014.12
Note: checking out 'arc-2014.12'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at eda052d... Check in ARCompact simulator.  A valid configuration is arc-elf. This is not quite finished and has most likely a few files that are obsolete & not used, but it's good enough to run gcc regression tests.
  pulling latest version
fatal: No remote repository specified.  Please, specify either a URL or a
remote name from which new revisions should be fetched.
```

**Comment by anthony-kolesov** [comment]
Hi Calvin,

No, tags haven't been changed. I've tried to reproduce this on my side, but everything works.

In the end of your all-build.log there is "pulling latest version" - that shouldn't be there, because when tag has been checked out repository is in "detached" state and cannot be pulled - therefore you get an error. By default --auto-pull is enabled (and your options do not disable it), however our checkout script "arc-versions.sh" checks whether repository is in detached state or not (https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-4.8-dev/arc-versions.sh#L150). So we do this:

```
if ! git branch | grep -q -e '\* (detached from .*)' -e '\* (no branch)'
then
    echo "  pulling latest version"
    if ! git pull
    then
        exit 1
    fi
fi
```

In you cause what _should_ happen is that grep should find "detached from", return zero status, which is inverted and branch is not taken. However in real life for your it happens otherwise. 

After some further investigation I can assume that your system has Git 2.4, which prints a third kind of message:

```
* (HEAD detached at arc-2014.12)
```

I will add support for those messages for the next release, however I cannot modify tag 2014.12 itself.

**Comment by calvinatintel** [comment]
Thank you as always for your quick and accurate diagnosis and responses.
We have fixed the `2014.12` tag build by patching the `arc-versions.sh` with your change:

```
cd toolchain
wget https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/commit/0664fe50331e30e0d62cc7dd1b81cbb29150526d.patch
sed -i 's|156,|167,|g' 0664fe50331e30e0d62cc7dd1b81cbb29150526d.patch
patch --ignore-whitespace -p1 < 0664fe50331e30e0d62cc7dd1b81cbb29150526d.patch
./arc-clone-all.sh
./build-all.sh
```
